app/controllers/articles_controller.rb

```diff
@@ -268,6 +267,15 @@ def article_params_json
     params.require(:article).permit(allowed_params)
   end
 
+  def handle_notifications(updated)
+    if updated && @article.published && @article.saved_changes["published"] == [false, true]
+      Notification.send_to_followers(@article, "Published")
+    elsif @article.saved_changes["published"] == [true, false]
+      Notification.remove_all_without_delay(notifiable_id: @article.id, notifiable_type: "Article", action: "Published")
+      Notification.remove_all(notifiable_id: @article.id, notifiable_type: "Article", action: "Reaction")
```

Comment:
Should we consider also removing article's comment notification? I could be overthinking this.